docs/proposals/config-management-plugin-v2.md

```diff
@@ -0,0 +1,290 @@
+This selection is based on the recognition of well-known files in the directory (e.g. Chart.yaml, kustomization.yaml, etc...). 
+
+Currently, unlike natively supported tools, when a plugin is used, a user needs to explicitly specify the plugin 
+that should be used to render the manifests. As part of the improvements to config management plugins, 
+
+We want to provide the same ability to auto-select the plugin based on recognized files in the path of the git repository.
+
+### Non-Goals
+
+#### Parameters support in UI/CLI
+Currently, configuration management plugins allow specifying only a list of environment variables via UI/CLI. 
+
+We want to extend its functionality to provide a similar experience as for existing natively supported tools 
+to additional config management tools. 
+
+## Proposal
+
+We have drafted the solution to the problem statement as **running configuration management plugin tools as sidecar in the argocd-repo-server**. 
+
+All it means that Argo CD Config Management Plugin 2.0 will be,
+- A user-supplied container image with all the necessary tooling installed in it. 
+- It will run as a sidecar in the repo server deployment and will have shared access to the git repositories.
+- It will contain a CMP YAML specification file describing how to render manifests.
+- Its entrypoint will be a lightweight CMP API server that receives requests by the main repo-server to render manifests, 
+based on the CMP specification file.
+
+This mechanism will provide the following benefits over the existing solution,
+- Plugin owners control their execution environment, packaging whatever dependent binaries required.
+- An  Argo CD user who wants to use additional config management tools does not have to go through the hassle of building 
+a customized argocd-repo-server in order to install required dependencies. 
+- The plugin image will be running in a container separate from the main repo-server.
+
+### Use cases
+
+* UC1: As an Argo CD user, I would like to use first-class support provided for additional tools to generate and manage deployable kubernetes manifests
+* UC2: As an Argo CD operator, I want to have smooth experience while installing additional tools such as  cdk8s, Tanka, jkcfg, QBEC, Dhall, pulumi, etc.
+* UC3: As a plugin owner, I want to have some control over the execution environment as I want to package whatever dependent binaries required. 
+
+### Implementation Details
+
+Config Management Plugin v2.0 implementation and experience will be as,
+
+#### Installation
+
+To install a plugin, an operator will simply patch argocd-repo-server to run config management plugin container as a sidecar, 
+with argocd-cmp-server as it’s entrypoint:
+
+```bash
+# A plugin is a container image which runs as a sidecar, with the execution environment
+# necessary to render manifests. To install a plugin, 
+containers:
+- name: cdk8s
+  command: [/var/run/argocd/argocd-cmp-server]
+  image: docker.ui/cdk8s/cdk8s:latest
```

Comment:
Could you please mention that this could be an off-the-shelf image too ?

Reply:
will do.